exec/java-exec/src/main/java/org/apache/drill/exec/planner/common/DrillJoinRelBase.java

```diff
@@ -209,4 +210,36 @@ public static RelOptCost computeHashJoinCostWithRowCntKeySize(RelOptPlanner plan
     return costFactory.makeCost(buildRowCount + probeRowCount, cpuCost, 0, 0, memCost);
   }
 
+  @Override public boolean isValid(Litmus litmus, Context context) {
```

Comment:
I think this row type validation could be moved to the `RowKeyJoinRel`.   Doing this in the base class of the joins does not seem right considering the Calcite is already doing the check for SemiJoin in its `isValid()` implementation,